### ISSUE TITLE
API for adding new fields to `notebook_node` that provides js like dot access to the attributes 

### ISSUE THREAD

**Issue by mpacer** [jupyter/nbformat#102]
The dot access method made available by `Notebook_Node` instances is convenient for concisely expressing nested calls into a notebook structure. This kind of concision is key to having understandable templates in `nbconvert`.

@minrk proposes on [nbconvert](https://github.com/jupyter/nbconvert/pull/643#issuecomment-321224684) that we use an intermediate representation that is allowed to add fields to the notebook (at least in the `cell.transient` namespace). But currently the obvious way to add values to that non-existant transient field is to assign a dict to it:

i.e., 
```python
cell.transient = {
    'remove_source': True
}
```

But, that results in a transient field with a dictionary type value that doesn't similarly have dot access to its values so `cell.transient` works but `cell.transient.remove_source` returns an `AttributeError` (specifically `*** AttributeError: 'dict' object has no attribute 'remove_source'`). 

Can someone point me to how this dot access is managed?

I'll make a PR so that we can have an API along the lines of `cell.add_field("transient", dict_like_object)` where `dict_like_object = {"remove_source": True}`. This would then create & recursively add `transient` and dict-like objects below it as a `NotebookNode` objects so that this syntactic convenience can be maintained.

**Comment by takluyver** [comment]
I think this should do it:

```python
from nbformat import from_dict

cell.transient = from_dict({
    'remove_source': True
})
```

I would say there's no need for a dedicated `add_field` method - it's another thing to understand, and it doesn't save any typing.

**Comment by minrk** [comment]
Yup, from_dict already addresses this pretty nicely, I think.

**Comment by minrk** [comment]
We could also potentially call from_dict internally if `setattr` is called with a dict, to ensure that it stays nodes all the way down.

**Comment by mpacer** [comment]
If there's support for the latter (re: `setattr` →`from_dict`) I'm willing to take a stab at implementing it.

**Comment by minrk** [comment]
@mpacer I think it makes sense if you'd like to give it a go. It should be pretty small.

**Comment by mpacer** [comment]
Closed by #107.
